cloudahoywriter.lua

```diff
@@ -375,6 +409,42 @@ function maybe_write_data()
     lastWriteTime = os.clock()
 end
 
+-- Called from show_ui() on every draw. Keep this fast.
+-- Decides whether to automatically start or stop recording.
+function automatic_recording_state_check()
+    if CAWR_isPaused == 1 then return end
+
+    if not CAWR_groundSpeed then
+        return -- Possible race with registering datarefs
+    end
+
+    local currentTime = os.time()
```

Comment:
Should we be using system time for these timeouts or simulation time?

Reply:
Nice catch. Yes, switching to simulation time.